src/memory.rs

```diff
@@ -7,34 +7,31 @@ use ic_stable_structures::StableBTreeMap;
 use ic_stable_structures::VectorMemory;
 use std::cell::RefCell;
 
-use crate::types::{ApiKey, Metrics, PrincipalStorable, ProviderId};
+use crate::types::{ApiKey, Metrics, ProviderId};
 
 #[cfg(not(target_arch = "wasm32"))]
 type Memory = VirtualMemory<VectorMemory>;
 #[cfg(target_arch = "wasm32")]
 type Memory = VirtualMemory<DefaultMemoryImpl>;
 
 thread_local! {
-    // Unstable static data: this is reset when the canister is upgraded.
+    // Unstable static data: these are reset when the canister is upgraded.
     pub static UNSTABLE_METRICS: RefCell<Metrics> = RefCell::new(Metrics::default());
 
-    // Stable static data: this is preserved when the canister is upgraded.
+    // Stable static data: these are preserved when the canister is upgraded.
     #[cfg(not(target_arch = "wasm32"))]
     static MEMORY_MANAGER: RefCell<MemoryManager<VectorMemory>> =
         RefCell::new(MemoryManager::init(VectorMemory::new(RefCell::new(vec![]))));
     #[cfg(target_arch = "wasm32")]
     static MEMORY_MANAGER: RefCell<MemoryManager<DefaultMemoryImpl>> =
         RefCell::new(MemoryManager::init(DefaultMemoryImpl::default()));
-    static API_KEY_PRINCIPALS: RefCell<Vec<PrincipalStorable>> = RefCell::new(vec![]);
+    static API_KEY_PRINCIPALS: RefCell<Vec<Principal>> = RefCell::new(vec![]);
```

Comment:
I think here we should use an `ic_stable_structures::Vec` and not a regular `std::vec::Vec`

Reply:
Good catch; I didn't know there was a distinction. It seems like `BoundedStorable` isn't implemented for `Principal`, so it seems like I would have to reintroduce the `PrincipalStorable` struct to be able to use this. Does that sound right?

Reply:
There's a distinction between `Storable` and `BoundedStorable`, where `Principal` implements the first but not the second for some reason. This should probably be added to the `ic-stable-structures` crate since it would be so simple and unopinionated:

```rust
impl BoundedStorable for Principal {
    const MAX_SIZE: u32 = 29;
    const IS_FIXED_SIZE: bool = false;
}
```

Reply:
Note that there is no `BoundedStorable` anymore since [v0.6.0](https://github.com/dfinity/stable-structures/releases/tag/v0.6.0).
> chore: merge BoundedStorable into Storable by @ielashi in https://github.com/dfinity/stable-structures/pull/94

This has been replaced by `const BOUND: Bound = Bound::Bounded`, which is in particular implemented for `Principal`

Reply:
Okay, good to know. That design makes a lot more sense. Last time I tried to update to the latest IC release (a few months ago), the blocker was a conflict with the Candid version. It might be possible to update to the last version, but I'll do this in a separate PR since it will be nontrivial.